src/spi_flash.c

```diff
@@ -739,9 +739,28 @@ int Drv_SpiInitializeFlash(const FLASHER_SPI_CONFIGURATION_T *ptSpiCfg, FLASHER_
 				/* yes, detected spi flash -> copy all attributes */
 				memcpy(&ptFlash->tAttributes, ptFlashAttr, sizeof(SPIFLASH_ATTRIBUTES_T));
 
+				unsigned short manualSpeed = 0;
+				/* If the passed speed was set manually use the passed value instead of the detected value */
+				if(ptSpiCfg->ulInitialSpeedKhz != 1000 || ptSpiCfg->ulMaximumSpeedKhz != 25000)
```

Comment:
This check is odd - if we do it like this, you won't be able to pass other values than the 1000 and 25000 kHz which are currently set in flasher.lua. The meaning of the ulMaximumSpeedKHz parameter is changed here.
Maybe it's better to extend the structure and add a new parameter, ulSpeedKhz, which is ignored when 0.